src/core/pointcloud/expression/qgspointcloudexpressionnodeimpl.h

```diff
@@ -0,0 +1,368 @@
+      uoMinus,
+    };
+
+    /**
+     * A node unary operator is modifying the value of \a operand by negating it with \a op.
+     */
+    QgsPointCloudExpressionNodeUnaryOperator( QgsPointCloudExpressionNodeUnaryOperator::UnaryOperator op, QgsPointCloudExpressionNode *operand SIP_TRANSFER )
+      : mOp( op )
+      , mOperand( operand )
+    {}
+    ~QgsPointCloudExpressionNodeUnaryOperator() override { delete mOperand; }
+
+#ifdef SIP_RUN
+    SIP_PYOBJECT __repr__();
+    % MethodCode
+    QString str = QStringLiteral( "<QgsPointCloudExpressionNodeUnaryOperator: %1>" ).arg( sipCpp->text() );
+    sipRes = PyUnicode_FromString( str.toUtf8().constData() );
+    % End
+#endif
+
+    /**
+     * Returns the unary operator.
+     */
+    QgsPointCloudExpressionNodeUnaryOperator::UnaryOperator op() const { return mOp; }
+
+    /**
+     * Returns the node the operator will operate upon.
+     */
+    QgsPointCloudExpressionNode *operand() const { return mOperand; }
+
+    QgsPointCloudExpressionNode::NodeType nodeType() const override;
+    bool prepareNode( QgsPointCloudExpression *parent, const QgsPointCloudBlock *block ) override;
+    double evalNode( QgsPointCloudExpression *parent, int pointIndex ) override;
+    QString dump() const override;
+
+    QSet<QString> referencedAttributes() const override;
+    QList<const QgsPointCloudExpressionNode *> nodes() const override; SIP_SKIP
+    QgsPointCloudExpressionNode *clone() const override SIP_FACTORY;
+
+    bool isStatic( QgsPointCloudExpression *parent, const QgsPointCloudBlock *block ) const override;
+
+    /**
+     * Returns a the name of this operator without the operands.
+     * I.e. "NOT" or "-"
+     */
+    QString text() const;
+
+    static bool convert( const QgsExpressionNodeUnaryOperator::UnaryOperator source, QgsPointCloudExpressionNodeUnaryOperator::UnaryOperator &target );
+
+  private:
+    UnaryOperator mOp;
+    QgsPointCloudExpressionNode *mOperand = nullptr;
+
+    static const char *UNARY_OPERATOR_TEXT[];
+};
+
+/**
+ * \brief A binary expression operator, which operates on two values.
+ * \ingroup core
+ *
+ * \since QGIS 3.26
+ */
+class CORE_EXPORT QgsPointCloudExpressionNodeBinaryOperator : public QgsPointCloudExpressionNode
+{
+  public:
+
+    /**
+     * \brief list of binary operators
+     * \note if any change is made here, the definition of QgsPointCloudExpression::BinaryOperatorText[] must be adapted.
+     */
+    enum BinaryOperator
+    {
+      // logical
+      boOr,
+      boAnd,
+
+      // comparison
+      boEQ,  //!< =
+      boNE,  //!< <>
+      boLE,  //!< <=
+      boGE,  //!< >=
+      boLT,  //!< <
+      boGT,  //!< >
+
+      // math
+      boPlus,
+      boMinus,
+      boMul,
+      boDiv,
+      boIntDiv,
+      boMod,
+      boPow,
+
+      // other operators not implemented in pointcloud expressions
+      boRegexp,
+      boLike,
+      boILike,
+      boNotLike,
+      boNotILike,
+      boIs,
+      boIsNot,
+      boConcat,
```

Comment:
Sorry I think I didn't explain correctly what I meant... The idea was that that operators that are not supported in pointcloud expressions would not be listed in this enum at all - we do not really have any use for these unsupported enum values...